src/core/varlink-common.c

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "json-util.h"
+#include "rlimit-util.h"
+#include "varlink-common.h"
+
+int rlimit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        const struct rlimit *rl = userdata;
+        struct rlimit buf = {};
+        int r;
+
+        assert(ret);
+        assert(name);
+
+        if (!rl) {
+                const char *p;
+                int z;
+
+                /* Skip over any prefix, such as "Default" */
+                assert_se(p = strstrafter(name, "Limit"));
+
+                z = rlimit_from_string(p);
+                assert(z >= 0 && z < _RLIMIT_MAX);
+
+                r = getrlimit(z, &buf);
```

Comment:
I guess this comes from `bus_property_get_rlimit`? This is iffy and would cause issues like #34193. It's probably OK to use in `Manager`, but not in `ExecContext`. So please move this into `manager-varlink.c` and add a comment that we use current process's value as fallback.

(It's on my TODO list to fix the dbus property getter)

Reply:
FTR: https://github.com/systemd/systemd/pull/35238 uses `rlimit_build_json()` in ExecContext

Reply:
@YHNdnzj I've looked once again into this. Here are some thoughts. Note that they also referring [this part](https://github.com/systemd/systemd/pull/35238/files#diff-08cbab828d752d560bbc8800b270a61292c5f758b9f85d550d298000d54d6628R1068) of https://github.com/systemd/systemd/pull/35238:
1. `rlimit_build_json()` below does `if (rl->rlim_cur == RLIM_INFINITY && rl->rlim_max == RLIM_INFINITY) return 0;` which means that unset limit will be skipped from the output instead of showing "infinity".
2. I'm falling back to default rlimit in exec context (via `RLIMIT_WITH_DEFAULT` helper). It should address your concern described in #34193

Please let me know if you think otherwise.